pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -290,6 +300,28 @@ private String extractHttpHeader(HttpHeaders headers, String name) {
     return value;
   }
 
+  private Pair<Boolean, String> encryptSegmentIfNeeded(String offlineTableName, File tempDecryptedFile, File tempEncryptedFile,
+      boolean uploadedSegmentIsEncrypted) {
+
+    // get crypter class name from table config
+    String crypterClassName = _pinotHelixResourceManager.getCrypterClassNameFromTableConfig(offlineTableName);
+
```

Comment:
How about returning right here if segment does not need encryption? The logic is easier to follow then.

Reply:
Done.

Reply:
IMO, its better to always go through the same path and just make the crypter no-op